core/string/ustring.cpp

```diff
@@ -541,40 +541,41 @@ bool String::operator==(const wchar_t *p_str) const {
 }
 
 bool String::operator==(const char32_t *p_str) const {
-	const int len = strlen(p_str);
+	if (p_str == ptr()) {
```

Comment:
```suggestion
	if (unlikely(p_str == ptr())) {
```

Reply:
This case only makes sense if you for some reason do `my_str_1 == my_str_2.ptr()` or similar, I don't think this one makes sense